src/executor.rs

```diff
@@ -119,18 +132,44 @@ impl Executor {
 
         while let Some(cmd) = self.cmd_rx.recv().await {
             match cmd {
-                ExecutorCommand::BlockReceived {
+                ExecutorCommand::BlockReceivedGossip {
                     block,
                 } => {
-                    log::info!(
-                        "ExecutorCommand::BlockReceived height={}",
-                        block.header.height
-                    );
-                    self.alert_tx.send(Alert::BlockReceived {
+                    self.alert_tx.send(Alert::BlockReceivedGossip {
                         block_height: block.header.height.parse::<u64>()?,
                     })?;
                     self.execute_block(*block).await?;
                 }
+                #[cfg(feature = "reader")]
+                ExecutorCommand::BlockReceivedFromDA {
+                    block,
+                } => {
+                    self.alert_tx.send(Alert::BlockReceivedFromDA {
+                        block_height: block.header.height.parse::<u64>()?,
+                    })?;
+
+                    match self.sequencer_hash_to_execution_hash.get(&block.block_hash) {
+                        Some(execution_block_hash) => {
+                            self.execution_rpc_client
+                                .call_finalize_block(execution_block_hash.clone())
+                                .await?;
```

Comment:
updated so that the error is just logged. however, there are probably some cases where taking the whole program down is okay, for example in the case of programmer error (we try to finalize a block that hasn't been executed)